src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/ApplicationInsights/DefaultTelemetryClientFactory.cs

```diff
@@ -92,5 +130,31 @@ internal static void AddInitializers(TelemetryConfiguration config)
             // This applies our special scope properties and gets RoleInstance name
             config.TelemetryInitializers.Add(new WebJobsTelemetryInitializer());
         }
+
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Disposes the instance.
+        /// </summary>
+        /// <param name="disposing"></param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing && !_disposed)
+            {
+                // TelemetryConfiguration.Dispose will dispose the Channel and the TelemetryProcessors
+                // registered with the TelemetryProcessorChainBuilder.
+                _config.Dispose();
```

Comment:
make these `?.Dispose` to help avoid issues with derived classes (these should override dispose anyway but..)